packages/wdio-cli/src/interface.js

```diff
@@ -0,0 +1,155 @@
+import chalk from 'chalk'
+import cliSpinners from 'cli-spinners'
+import EventEmitter from 'events'
+import CLIInterface from 'wdio-interface'
+
+const clock = cliSpinners['clock']
+
+export default class WDIOCLIInterface extends EventEmitter {
+    constructor (config, specs) {
+        super()
+        this.hasAnsiSupport = chalk.supportsColor.hasBasic
+        this.clockTimer = 0
+        this.specs = specs
+        this.config = config
+        this.jobs = new Map()
+        this.start = Date.now()
+        this.result = {
+            finished: 0,
+            passed: 0,
+            failed: 0
+        }
+        this.messages = {
+            reporter: {}
+        }
+
+        this.interface = new CLIInterface()
+        this.on('job:start', ::this.addJob)
+        this.on('job:end', ::this.clearJob)
+    }
+
+    /**
+     * add job to interface
+     */
+    addJob({ cid, caps, specs }) {
+        this.jobs.set(cid, { caps, specs })
+        this.updateView()
+    }
+
+    /**
+     * clear job from interface
+     */
+    clearJob ({ cid, passed }) {
+        this.jobs.delete(cid)
+        this.result.finished++
+
+        if (passed) {
+            this.result.passed++
+        } else {
+            this.result.failed++
+        }
+
+        this.updateView(true)
+    }
+
+    /**
+     * event handler that is triggered when runner sends up events
+     */
+    onMessage (params) {
+        if (!this.messages[params.origin][params.name]) {
+            this.messages[params.origin][params.name] = []
+        }
+        this.messages[params.origin][params.name].push(params.content)
+    }
+
+    updateView (wasJobCleared) {
+        const isFinished = this.jobs.size === 0
+        const pendingJobs = this.specs.length - this.jobs.size - this.result.finished
```

Comment:
Is this correct?

If 8 out of 10 jobs/specs have been run, the logic would be that there are 0 pending jobs if I am not mistaken as the math would be like the following, correct? 10 - 2 - 8 = 0

Reply:
This should be correct, here is what each subtrahend means:

Reply:
It should, here is what each subtrahend means:
- `this.specs.length`: total amount of specs
- `this.jobs.size`: current running jobs
- `this.result.finished`: finished jobs

If there are 10 total jobs, two are running and 8 are finished there are no jobs that have to get scheduled anymore.